tests/scripts/test_labels.py

```diff
@@ -0,0 +1,226 @@
+        'short-diamondC_1x1x1_pp-vmc-estimator-energydensity-cell',
+        'short-diamondC_1x1x1_pp-dmc-estimator-energydensity-cell',
+
+        # https://github.com/QMCPACK/qmcpack/issues/1040
+        'short-diamondC-afqmc_hyb_nn2',
+        'short-diamondC-afqmc_incmf_nn2',
+        'short-diamondC-afqmc_nn2',
+        ])
+
+    # aos specific (but general otherwise)
+    if aos:
+        unsupported |= set()
+    #end if
+
+    # soa specific (but general otherwise)
+    if soa:
+        broken |= set()
+    #end if
+
+    # gpu specific (but general otherwise)
+    if gpu:
+        broken      |= set()
+        unsupported |= set()
+    #end if
+
+    # real specific (but general otherwise)
+    if real:
+        unsupported |= set()
+    #end if
+
+    # complex specific (but general otherwise)
+    if comp:
+        unsupported |= set()
+    #end if
+    
+    # mixed precision specific (but general otherwise)
+    if mixed:
+        broken      |= set()
+        unsupported |= set()
+    #end if
+
+    # finer details based on build
+    # AoS issues
+    if aos and cpu and real and full:
+        None
+    elif aos and cpu and real and mixed:
+        None
+    elif aos and cpu and comp and full:
+        None
+    elif aos and cpu and comp and mixed:
+        None
+    elif aos and gpu and real:
+        None
+    elif aos and gpu and comp:
+        None
+    # SoA issues
+    elif soa and cpu and real and full:
+        None
+    elif soa and cpu and real and mixed:
+        None
+    elif soa and cpu and comp and full:
+        None
+    elif soa and cpu and comp and mixed:
+        None
+    elif soa and gpu and real:
+        None
+    elif soa and gpu and comp:
+        None
+    #end if
+
+    code_fail = hard_fail | abort
+    stat_fail = def_stat_fail | int_stat_fail
+
+    bug = broken | code_fail | def_stat_fail
+
+    unstable = broken | unsupported | code_fail | stat_fail
+
+    label_sets = dict(
+        unstable               = unstable,
+        broken                 = broken,
+        unsupported            = unsupported,
+        hard_fail              = hard_fail,
+        abort                  = abort,
+        definite_stat_fail     = def_stat_fail,
+        intermittent_stat_fail = int_stat_fail,
+        code_fail              = code_fail,
+        stat_fail              = stat_fail,
+        bug                    = bug,
+        )
+
+    return label_sets
+#end def create_label_sets
+
+
+# extract test name and build flags from args
+try:
+    full_test,qmc_cuda,enable_soa,qmc_complex,qmc_mixed = sys.argv[1:]
+    qmc_cuda    = qmc_cuda=='1'
+    enable_soa  = enable_soa=='1'
+    qmc_complex = qmc_complex=='1'
+    qmc_mixed   = qmc_mixed=='1'
+    cpu   = not qmc_cuda
+    gpu   = qmc_cuda
+    aos   = not enable_soa
+    soa   = enable_soa
+    real  = not qmc_complex
+    comp  = qmc_complex
+    full  = not qmc_mixed
+    mixed = qmc_mixed
+except:
+    error('extract')
```

Comment:
It might aid debugging if the exception handler printed the exception.

Reply:
It's true, but the output from this script is consumed directly by the ctest system (and not printed to user visible stdout), so I thought providing a label was a better way to go.

Reply:
It would be combined with checking the return value in the CMake script - if the script fails, print stdout.   Alternately, the exception handler could print to stderr and the CMake script could capture that and print that if the return code is nonzero.

Reply:
That's true and it would be an improvement.  I will add it.